source/Handlebars/HtmlEncoder.cs

```diff
@@ -11,9 +11,36 @@ public string Encode(string text)
             if (text == null)
                 return String.Empty;
 
-            var sb = new StringBuilder(text.Length);
 
+            // Detect if we need to allocate a stringbuilder and new string
             for (var i = 0; i < text.Length; i++)
+            {
+                switch (text[i])
+                {
+                    case '"':
+                    case '&':
+                    case '<':
+                    case '>':
+                        return ReallyEncode(text, i);
+                    default:
+                        if (text[i] > 159)
+                        {
+                            return ReallyEncode(text, i);
+                        }
+                        else
```

Comment:
@304NotModified Could this else be omitted? Or, at the very least be use braces `{}` to match the style of the rest of the project?